# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegDataDecoder.h
# Commit: a599c79ed651
# Full Hash: a599c79ed65176fd4e66394d1144f61931def4c9
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-03-04 16:19:40
# Description:
#   Bug 1619544 [Wayland] Do locking in VA-API codec creation, r=jya
#   
#   We need to lock codec creation as ffmpeg/libva does not provide any kind of synchronization here.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65089
# ==============================================================================

diff -r bf8c2eefd9b2 -r a599c79ed651 dom/media/platforms/ffmpeg/FFmpegDataDecoder.h
--- a/dom/media/platforms/ffmpeg/FFmpegDataDecoder.h	Wed Mar 04 07:13:47 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegDataDecoder.h	Wed Mar 04 07:09:47 2020 +0000
@@ -59,6 +59,9 @@
   RefPtr<MediaByteBuffer> mExtraData;
   AVCodecID mCodecID;
 
+ protected:
+  static StaticMutex sMonitor;
+
  private:
   RefPtr<DecodePromise> ProcessDecode(MediaRawData* aSample);
   RefPtr<DecodePromise> ProcessDrain();
@@ -71,7 +74,6 @@
   const RefPtr<TaskQueue> mTaskQueue;
   MozPromiseHolder<DecodePromise> mPromise;
   media::TimeUnit mLastInputDts;
-  static StaticMutex sMonitor;
 };
 
 }  // namespace mozilla